src/api_server/src/parsed_request.rs

```diff
@@ -294,16 +294,47 @@ pub(crate) mod tests {
         }
     }
 
+    fn http_response(body: &str, status_code: i32) -> String {
+        let header = format!(
+            "HTTP/1.1 {} \r\n\
+             Server: Firecracker API\r\n\
+             Connection: keep-alive\r\n",
+            status_code
+        );
+
+        let content = format!(
+            "Content-Type: application/json\r\n\
+             Content-Length: {}\r\n\r\n{}",
+            body.len(),
+            body,
+        );
+
+        if status_code == 204 {
+            // No Content
+            return format!("{}{}", header, "\r\n");
+        }
+        format!("{}{}", header, content)
```

Comment:
No point in building `content` if not used, also the preferred style for conditional returning is `if/else` block:
```suggestion
        if status_code == 204 {
            // No Content
            format!("{}\r\n", header)
        } else {
            let content = format!(
                "Content-Type: application/json\r\n\
                 Content-Length: {}\r\n\r\n{}",
                body.len(),
                body,
            );
            format!("{}{}", header, content)
        }
```